packages/preview/touying-unistra-pristine/1.0.0/src/unistra.typ

```diff
@@ -0,0 +1,856 @@
+#let slide(
+  config: (:),
+  repeat: auto,
+  setting: body => body,
+  composer: auto,
+  ..bodies,
+) = touying-slide-wrapper(self => {
+  let footer(self) = {
+    let cell(body) = rect(
+      width: 150%,
+      height: 100%,
+      inset: 0mm,
+      outset: 0mm,
+      fill: none,
+      stroke: none,
+      align(horizon + center, text(size: 0.5em, fill: self.colors.black, body)),
+    )
+
+    set align(center + horizon)
+
+    let has-title-and-subtitle = (
+      self.info.title,
+      self.info.subtitle,
+    ).all(x => x not in ("", none))
+
+    block(
+      width: 125%,
+      height: 1.8em,
+      outset: 2mm,
+      stroke: (top: 0.5pt + self.colors.black),
+      {
+        set text(size: 1.5em)
+        grid(
+          columns: (auto, auto, auto),
+          rows: (1.4em, 1.4em),
+          gutter: 3pt,
+          cell(image("../assets/unistra.svg", width: auto, height: 100%)),
+          cell(
+            box(
+              text(
+                self.info.title,
+                weight: "bold",
+              ) + if has-title-and-subtitle and settings.FOOTER-SHOW-SUBTITLE {
+                settings.FOOTER-UPPER-SEP
+              } else {
+                ""
+              } + if settings.FOOTER-SHOW-SUBTITLE {
+                self.info.subtitle
+              } + "\n" + utils.call-or-display(
+                self,
+                [#self.info.author | #self.info.date],
+              ),
+              width: 125%,
+            ),
+          ),
+          cell(
+            utils.call-or-display(
+              self,
+              context utils
+                .slide-counter
+                .display() + " / " + utils.last-slide-number,
+            ),
+          ),
+        )
+      },
+    )
+  }
+
+  let self = utils.merge-dicts(
+    self,
+    config-page(
+      header: if settings.SHOW-HEADER {
+        unistra-nav-bar(self)
+      } else {
+        none
+      },
+      footer: if settings.SHOW-FOOTER {
+        footer
+      } else {
+        none
+      },
+    ),
+  )
+
+  touying-slide(
+    self: self,
+    config: config,
+    repeat: repeat,
+    setting: setting,
+    composer: composer,
+    ..bodies,
+  )
+})
+
+/// Creates a title slide with a logo, title, subtitle, author, and date.
+///
+/// Example:
+///
+/// ```typst
+/// #title-slide(title: "Override Title", logo-width: 60%)[]
+/// ```
+///
+/// If no title or subtitle is provided, will use the info object.
+///
+/// - `title` (str): The title of the slide. Default: "".
+///
+/// - `subtitle` (str): The subtitle of the slide. Default: "".
+///
+/// - `logo-path` (str): The path to the logo. Default: "../assets/unistra.svg".
+///
+/// - `logo-width` (str): The width of the logo. Default: "40%".
+///
+/// - `logo-height` (str): The height of the logo. Default: "auto".
+///
+/// - `..args`: Additional arguments to pass to the slide.
+#let title-slide(
+  title: "",
+  subtitle: "",
+  logo-path: "../assets/unistra.svg",
```

Comment:
The logo-path will unfortunately not work with user images because of the way Typst currently resolves path. To fix this, you would need to accept an already loaded `image`, e.g. `logo: image("../assets/unistra.svg")`. Then users can provide their own image. To change the image settings, you can use a a set rule, but the `logo-width` and `logo-height` arguments would just become superfluous.

Reply:
I have changed the way images are loaded so that:
1) `hero` and `gallery` require inline images to avoid issues with paths,
2) `title-slide` now uses the value of the `logo` parameter in Touying's `config-info` in `show: unistra-theme.with` (it can be overriden)